### PR TITLE
Issue 238 doc finite volumes

### LINKED ISSUE #238: example notebooks: Finite Volumes and add a discretisation
Notebooks/tutorials so that Jamie can start looking into adding his discretisations.
- What to add in a discretisation (including a test for each): divergence, gradient, etc
- What finite volumes discretisation does for each of these (with diagrams?) for reference

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Add tutorial for how to add a spatial method, and example notebook for Finite Volumes. 
Please point out anything I've missed!
Fixes #238 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tinosulzer** [comment]
Have now added the Finite Volume notebook so this should be almost ready to show to Jamie and Ivan

File: examples/notebooks/spatial_methods/finite-volumes.ipynb [hunk 1]
```diff
@@ -0,0 +1,1102 @@
...
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Set up \n",
+    "macroscale = [\"negative electrode\", \"separator\", \"positive electrode\"]\n",
+    "x_var = pybamm.SpatialVariable(\"x\", domain=macroscale)\n",
+    "r_var = pybamm.SpatialVariable(\"r\", domain=[\"negative particle\"])\n",
+    "\n",
+    "# Set up \n",
```
Comment (rtimms) [inline examples/notebooks/spatial_methods/finite-volumes.ipynb]: Think this is a repeat of the above?

File: examples/notebooks/spatial_methods/finite-volumes.ipynb [hunk 2]
```diff
@@ -0,0 +1,1102 @@
...
+    "w = pybamm.Variable(\"w\") # w is a variable without a domain (e.g. time, average concentration)\n",
+    "\n",
+    "variables = [u,v,w]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before discretising, we trying to evaluate the variables raises a `NotImplementedError`:"
```
Comment (rtimms) [inline examples/notebooks/spatial_methods/finite-volumes.ipynb]: Typo here - remove the we

File: examples/notebooks/spatial_methods/finite-volumes.ipynb [hunk 3]
```diff
@@ -0,0 +1,1102 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([[-1.,  1.,  0., ...,  0.,  0.,  0.],\n",
```
Comment (rtimms) [inline examples/notebooks/spatial_methods/finite-volumes.ipynb]: Should there be a 1/dx out the front here?

**Pull request merged**